svc-systems/systems/storage.go

```diff
@@ -217,6 +217,13 @@ func (e *ExternalInterface) validateProperties(request *smodel.Volume) (int32, s
 				log.Printf(err.Error())
 				return http.StatusNotFound, response.ResourceNotFound, []interface{}{"Drives", driveURI}, fmt.Errorf("Error while getting drive details for %s", driveURI)
 			}
+			// Validating if a a drive URI contains correct system id
+			driveURI_split := strings.Split(driveURI, "/")
+			if driveURI_split[4] != systemId {
```

Comment:
Add a check for length of the returned array.

Reply:
Added now